lib/cancan/controller_resource.rb

```diff
@@ -228,11 +228,19 @@ def name
       @name || name_from_controller
     end
 
+    def strong_parameters?
+      @params.class.name == 'ActionController::Parameters'
+    end
+
     def resource_params
-      if @options[:class]
-        @params[@options[:class].to_s.underscore.gsub('/', '_')]
-      else
-        @params[namespaced_name.to_s.underscore.gsub("/", "_")]
+      if [:create, :update].member? @params[:action].to_sym
+        param_name = @options[:instance_name] || (@options[:class] || namespaced_name).to_s.underscore.gsub('/', '_')
+        if strong_parameters? || @options[:params]
+          params_method = (@options[:params] == true || ! @options[:params]) ?
+            "#{param_name}_params" : @options[:params]
+          return @controller.send params_method if @controller.send :respond_to?, params_method
```

Comment:
In order to make the param method private, [#respond_to?](http://ruby-doc.org/core-1.9.3/Object.html#method-i-respond_to-3F) should take `true` as a second param.